### ISSUE TITLE
Allow for custom charts to be stored in application repository and linked via relative path in bedrock.yaml

### ISSUE THREAD

**Issue by mtarng** [microsoft/bedrock#805]
**As a:**
SPK user

**I want:**
To be able to store helm charts in my application repository and have the components copied over to my HLD during the lifecycle pipeline.

**So that:**
I can couple incremental chart changes with my application changes in the same set of commits.

**Describe the solution you'd like:**

- [x] bedrock.yaml to accept strictly a path (not a git and a path) for helm config
- [x] lifecycle pipeline to trigger on a components directory (that stores helm charts) at the root of the application repo and copy over the components directory into HLD repository 
- [x] `spk hld reconcile` on a service will scaffold with reference (with relative path from leaf to root) the component/chart copied over and referenced in `bedrock.yaml`
- [x] Support git flow branches for ring names and normalize them in the HLD ie: `release/qa`, `release/dev`, `release/prod` should output `release-qa`, `release-dev`, and `release-prod` in the HLD.

1/24/20:
- [ ] Unit Tests for Bedrock.yaml with _only_ a path for helm chart
- [ ] Unit Tests for hld reconcile with bedrock.yaml containing _only_ a relative path for helm chart
- [x] Rebase with current master branch
- [ ] Find a better solution for the `../../../` path for helm charts, perhaps a top level well-known charts directory (eg: `charts`)

**Acceptance Criteria:**

bedrock.yaml
```
...
rings:
  qa:
  test:
  prod:
services:
  ./myservice
    helm:
      path: ./components/container
...
```

component.yaml
```
name: hldname
subcomponents:
- name: servicename
  type: component
  source: ./components/myservicename
```

TBD: other component.yaml to be added

**Describe alternatives you've considered:**

None

**Additional context:**

I want to be able consolidate my helm charts with my sources, such that changes on code and changes to infrastructure both trigger a update to my HLD. 

**Does this require updates to documentation?:**

yes

**Comment by mtarng** [comment]
Work is being done and documented in https://github.com/CatalystCode/spk/tree/charts-in-app-repo-805 as a feature branch until we're sure of the behavior we want.

**Comment by andrebriggs** [comment]
@evanlouie looks like this is done from the task list checkboxes. Any idea on the state of this? It was planned but is referenced from an epic #838 cc @bnookala @mtarng

**Comment by evanlouie** [comment]
I believe @bnookala was working on this but has not yet been merged, the work is in this branch: https://github.com/CatalystCode/spk/compare/charts-in-app-repo-805?expand=1

**Comment by andrebriggs** [comment]
What's the status of this? @bnookala @mtarng @evanlouie

**Comment by andrebriggs** [comment]
Marking this as on-hold @bnookala @mtarng @evanlouie

**Comment by bnookala** [comment]
presently, spk can consume charts from a separate repository, but cannot consume charts from an application/source repository. @andrebriggs

**Comment by mtarng** [comment]
Discussed with @andrebriggs, @bnookala , @evanlouie : The need can be covered with a simpler workflow via https://github.com/microsoft/bedrock/issues/956.

The consensus was that it's simpler to let fabrikate do the cloning of the charts when it needs them, so just provide access to the repositories where the charts live.

If, in the future, we need to cover the case where application, hld, and manifest repositories live in separate azdo orgs or live in separate azdo projects, we can revisit the workflow.

For now, this ticket can probably be marked "won't do".

**Comment by bnookala** [comment]
Closing this for now, as the work to support local charts has been solved by using the work from #956.

Continuing the work to support private helm charts in #914
